src/Computation/BitsExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Numerics;
+using Microsoft.Z3;
+using Symbolica.Expression;
+
+namespace Symbolica.Computation
+{
+    internal static class BitsExtensions
+    {
+        public static FPSort GetSort(this Bits self, Context context)
+        {
+            return (uint) self switch
+            {
+                16U => context.MkFPSort16(),
+                32U => context.MkFPSort32(),
+                64U => context.MkFPSort64(),
+                80U => context.MkFPSort(15U, 65U),
+                128U => context.MkFPSort128(),
+                _ => throw new Exception($"Floating-point size {self} is unsupported.")
+            };
+        }
+
+        public static BigInteger GetNan(this Bits self, Context context)
+        {
+            var sort = self.GetSort(context);
+            var nan = ((BigInteger.One << ((int) sort.EBits + 2)) - BigInteger.One) << ((int) sort.SBits - 2);
+
+            return ConstantUnsigned.Create(self, nan).Integer;
+        }
```

Comment:
The stuff above here is just moved from elsewhere since it made more sense to live here now that the other similar _filth_ is here.